app/src/main/java/com/github/se/cyrcle/permission/PermissionHandlerImpl.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.se.cyrcle.permission
+
+import android.content.pm.PackageManager
+import android.util.Log
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import com.github.se.cyrcle.MainActivity
+import com.mapbox.android.core.permissions.PermissionsListener
+import com.mapbox.android.core.permissions.PermissionsManager
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+
+/** Handles permissions for the app. */
+class PermissionHandlerImpl @Inject constructor() : PermissionHandler {
+
+  private lateinit var permissionManager: PermissionsManager
+  private lateinit var permissionLauncher: ActivityResultLauncher<Array<String>>
+
+  private val positionGranted: MutableStateFlow<Boolean> = MutableStateFlow(false)
+
+  /**
+   * Initializes the permissions handler.
+   *
+   * @param mainActivity the activity
+   */
+  override fun initHandler(mainActivity: MainActivity) {
+    positionGranted.value = PermissionsManager.areLocationPermissionsGranted(mainActivity)
+
+    // Create a PermissionsManager
+    permissionManager =
+        PermissionsManager(
+            object : PermissionsListener {
+              override fun onExplanationNeeded(permissionsToExplain: List<String>) {
+                Log.d("PermissionsHandlerImpl", "Permissions: $permissionsToExplain")
```

Comment:
You keep those log for debugging purpose ? Because I think it's good to keep them as it allows better tracking of the permission manager steps to process user permissions.

Reply:
Yes, it helps to follow the process of requesting permissions

Reply:
:+1: